ochazuke/webhooks/helpers.py

```diff
@@ -0,0 +1,239 @@
+    return mac.hexdigest()
+
+
+def signature_check(key, post_signature, payload):
+    """Check the HTTP POST legitimacy."""
+    if post_signature.startswith('sha1='):
+        sha_name, signature = post_signature.split('=')
+    else:
+        return False
+    if not signature:
+        return False
+    # HMAC requires its key to be bytes, but data is strings.
+    hexmac = get_payload_signature(key, payload)
+    return compare_digest(hexmac, signature.encode('utf-8'))
+
+
+def compare_digest(x, y):
+    """Create a hmac comparison.
+
+    Approximates hmac.compare_digest (Py 2.7.7+) until we upgrade.
+    TODO: SEE IF THIS NEEDS TO BE ADAPTED FOR PY3.6
+    """
+    if not (isinstance(x, bytes) and isinstance(y, bytes)):
+        raise TypeError("both inputs should be instances of bytes")
+    if len(x) != len(y):
+        return False
+    result = 0
+    for a, b in zip(bytearray(x), bytearray(y)):
+        result |= a ^ b
+    return result == 0
+
+
+def is_github_hook(request):
+    """Validate the github webhook HTTP POST request."""
+    if request.headers.get('X-GitHub-Event') is None:
+        return False
+    post_signature = request.headers.get('X-Hub-Signature')
+    if post_signature:
+        # TODO: DETERMINE WHERE TO HIDE SECRET KEY
+        key = app.config['HOOK_SECRET_KEY']
+        return signature_check(key, post_signature, request.data)
+    return False
+
+
+def extract_issue_event_info(payload):
+    """Extract information we need when handling webhook for issue events."""
+    # Extract the event-specific info and make our own timestamp
+    event_timestmp = datetime.utcnow().replace(microsecond=0).isoformat() + 'Z'
+    action = payload.get('action')
+    milestone = payload.get('issue')['milestone']
+    # If there is no milestone data, let title be None
+    if milestone is not None:
+        milestone_title = milestone.get('title')
+    else:
+        milestone_title = None
+    # Let details be None for opening/closing, but preserve old title on edits
+    if action in ['opened', 'closed', 'reopened', 'edited']:
+        changes = payload.get('changes')
+        if changes is not None:
+            details = {'old title': changes['title']['from']}
+        else:
+            details = None
+    elif action == ('milestoned' or 'demilestoned'):
+        details = {'milestone title': payload.get(
+            'issue')['milestone']['title']}
+    else:
+        details = {'label name': payload.get('label')['name']}
+    # Create a concise issue event dictionary
+    issue_event_info = {'issue_id': payload.get('issue')['number'],
+                        'header': payload.get('issue')['title'],
+                        'created_at': payload.get('issue')['created_at'],
+                        'milestone': milestone_title,
+                        'actor': payload.get('sender')['login'],
+                        'action': action,
+                        'details': details,
```

Comment:
details will contain contextual info depending on the action, or example previous milestone or title before it was edited.